common/scanmem.c

```diff
@@ -0,0 +1,1005 @@
+/*
+ *  Copyright (C) 2021 Cisco Systems, Inc. and/or its affiliates. All rights reserved.
+ *  Copyright (C) 2005-2010 Gianluigi Tiesi <sherpya@netfarm.it>
+ *
+ *  Authors: Gianluigi Tiesi
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ *  MA 02110-1301, USA.
+ */
+
+#include <windows.h>
+#include <tlhelp32.h>
+
+#include <psapi.h>
+#include <windns.h>
+
+#include <clamav.h>
+#include <others.h>
+
+#include "actions.h"
+#include "output.h"
+#include "exescanner.h"
+#include "scanmem.h"
+#include "clamdcom.h"
+
+typedef int (*proc_callback)(PROCESSENTRY32 ProcStruct, MODULEENTRY32 me32, void *data, struct mem_info *info);
+int sock;
+
+/* clamd helpers */
+static int chkpath(const char *path, struct mem_info *minfo)
```

Comment:
I'm curious if we can avoid having duplicate code by moving merging this and the one found in `clamdscan/proto.c`.  You will have to add at least one extra parameters to `chkpath()` to pass in `clamdopts`, and then change every use of `chkpath()` to pass that in.  I suppose the merged function will have to go somewhere under `common` but I'm not sure where. I suppose `misc.c`/`h` may be best - unless you have any better ideas. 🤷‍♀️